### PR TITLE
Mongodb identity and tests

### PR DESCRIPTION
**Pull request by enzo-ae**
As the current MongoDB generators use the EntityFramework with SQLite for identities and tests internally, I wanted to get rid of it and use only MongoDB.

This PR involves:
- Switch from EntityFrameworkCore.Identity (via SQLite) to AspNetCore.Identity.MongoDbCore
- Adaption of User and Role class
- Adaption of User service
- Override UserStore to track changes
- Switch Unit Tests to use MongoDb (Database gets cleaned before every Test Class)

Pretty sure that there are improvements (proper Session Management so that Test Classes don't have to run sequentially) but I think this is a good start.

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by nicolas63** [comment]
very nice work, and congratulations for you first contribution 👏

**Comment by nicolas63** [comment]
the integration tests failed do you need help to resolve the errors ?

**Comment by enzo-ae** [comment]
Thanks, I really enjoy the tool.

Would be great if you'd help me out as I don't know when I have the time to figure out how to run the tests locally in Win10 and fix the issues, might take a couple of days

**Comment by ivanmonteiro** [comment]
> Thanks, I really enjoy the tool.
> 
> Would be great if you'd help me out as I don't know when I have the time to figure out how to run the tests locally in Win10 and fix the issues, might take a couple of days

FYI, I usually when I want to reproduce an integration test error I do the following manual process:

First I need to npm link the jhipster project I'm working on (see https://github.com/jhipster/jhipster-dotnetcore#running-local-blueprint-version-for-development)

Then I copy those two files into the folder I just created (my-app if you followed the guide before):
 .yo-rc.json (see https://github.com/jhipster/jhipster-dotnetcore/tree/main/test-integration/samples)
and app.jdl (for the entities - see https://github.com/jhipster/jhipster-dotnetcore/tree/main/test-integration/samples/jdl-default)

And then I install jhipster and create the base application `jhipster --blueprint dotnetcore` and import the jdl `jhipster import-jdl app.jdl`. Now I can run tests locally, like `dotnet test` or `npm test`.

File: generators/server/templates/dotnetcore/test/Project.Test/Setup/TestStartup.cs.ejs [hunk 1]
```diff
@@ -35,8 +39,15 @@ namespace <%= namespace %>.Test.Setup
         private IConfiguration Configuration { get; }
 
         public override void Configure(IApplicationBuilder app, IHostEnvironment env, IServiceProvider serviceProvider,
-            ApplicationDatabaseContext context, IOptions<SecuritySettings> securitySettingsOptions)
+    <%_ if (databaseType === 'mongodb') { _%>
+            IMongoDatabaseContext context, IOptions<SecuritySettings> securitySettingsOptions)
         {
```
Comment (nicolas63) [inline generators/server/templates/dotnetcore/test/Project.Test/Setup/TestStartup.cs.ejs:44]: the problem is that the brace is not written when the database is != mongodb 

```suggestion
    {
    <%_ if (databaseType === 'mongodb') { _%>
            IMongoDatabaseContext context, IOptions<SecuritySettings> securitySettingsOptions)
        
```

File: generators/server/templates/dotnetcore/test/Project.Test/Setup/TestStartup.cs.ejs [hunk 2]
```diff
@@ -35,8 +39,16 @@ namespace <%= namespace %>.Test.Setup
...
         public override void Configure(IApplicationBuilder app, IHostEnvironment env, IServiceProvider serviceProvider,
+    {
+    <%_ if (databaseType === 'mongodb') { _%>
+            IMongoDatabaseContext context, IOptions<SecuritySettings> securitySettingsOptions)
+        
+            context.Database.ListCollectionNames()
+                .ForEachAsync(collection => context.Database.DropCollection(collection))
+                .Wait();
+    <%_ } else { _%>
             ApplicationDatabaseContext context, IOptions<SecuritySettings> securitySettingsOptions)
```
Comment (nicolas63) [inline generators/server/templates/dotnetcore/test/Project.Test/Setup/TestStartup.cs.ejs:50]: sorry is more like this

```suggestion    
    <%_ if (databaseType === 'mongodb') { _%>
            IMongoDatabaseContext context, IOptions<SecuritySettings> securitySettingsOptions)
            {
            context.Database.ListCollectionNames()
                .ForEachAsync(collection => context.Database.DropCollection(collection))
                .Wait();
    <%_ } else { _%>
            ApplicationDatabaseContext context, IOptions<SecuritySettings> securitySettingsOptions)
            {
```

**Comment by nicolas63** [comment]
@enzo-ae i think the namespace changes broke some components can you look the error ? or do you need help ?

**Comment by enzo-ae** [comment]
@nicolas63 Could you have a look at the entity server AutoMapperProfile? Haven't figured out why the maps are not created

**Comment by nicolas63** [comment]
close because #1092 integrate this changes

**Pull request closed without merging**